crates/client/src/client/schema/block.rs

```diff
@@ -75,10 +76,11 @@ pub struct BlockEdge {
     pub node: Block,
 }
 
-/// Block with transactiuon ids
+/// Block with transaction ids
 #[derive(cynic::QueryFragment, Clone, Debug)]
 #[cynic(schema_path = "./assets/schema.sdl")]
 pub struct Block {
+    pub version: Version,
```

Comment:
We have a separate enum with all supported versions for `ConsensusParameters`(and other parameters). This strategy forces us to cover all possible values during compilation time, and I think it is still good to continue using it.

<img width="526" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/9edc11f2-ffa7-4f1f-95b9-3b5b0c135c78">

Reply:
Updated to use `enum` at the `async-graphql` server level and at the `cynic` client level. This gives us exhaustive variant checks.